### PR TITLE
Document Heat removal in 2024.1 OpenStack upgrade

### PR DESCRIPTION
**Pull request by MoteHue**
I chose not to use any * shortenings, as we've been burnt by picking up unexpected systemd units before.

### CODE REVIEW

File: doc/source/operations/upgrading-openstack.rst [hunk 1]
```diff
@@ -52,7 +52,22 @@ driver. Instructions for enabling the driver can be found `here
...
+
+.. code-block:: console
+
+   kayobe overcloud host command run --command "systemctl stop kolla-heat_api-container.service kolla-heat_api_cfn-container.service kolla-heat_engine-container.service" -l controllers -b
+   kayobe overcloud host command run --command "systemctl disable kolla-heat_api-container.service kolla-heat_api_cfn-container.service kolla-heat_engine-container.service" -l controllers -b
+   kayobe overcloud host command run --command "rm /etc/systemd/system/kolla-heat_api-container.service" -l controllers -b
+   kayobe overcloud host command run --command "rm /etc/systemd/system/kolla-heat_api_cfn-container.service" -l controllers -b
+   kayobe overcloud host command run --command "rm /etc/systemd/system/kolla-heat_engine-container.service" -l controllers -b
+
+   kayobe overcloud host command run --command "docker rm heat_api heat_api_cfn heat_engine" -l controllers
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading-openstack.rst:73]: Do we want to remove the images as well?

**Review by priteau — CHANGES_REQUESTED** [review]
You probably want to remove the haproxy configuration and restart the haproxy container first, then stop/delete the heat containers.

**Review by priteau — CHANGES_REQUESTED** [review]
More is needed to be thorough. Remove Heat configuration generated by Kolla:

```
kayobe overcloud host command run --command "rm -rf /etc/kolla/heat-api /etc/kolla/heat-api-cfn /etc/kolla/heat-engine " --limit controllers -b
```

Remove the Heat service, endpoints (done automatically on service removal), user and domain. From an OpenStack virtual environment:

```
openstack service delete heat
openstack user delete heat
openstack domain set --disable heat_user_domain
openstack domain delete heat_user_domain
```

You can drop the database too, unless you want to keep historical content.

File: doc/source/operations/upgrading-openstack.rst (hunk 1 above)
Comment (priteau) [reply doc/source/operations/upgrading-openstack.rst:73]: We can recommend a `docker image prune -af` at the end.

Comment (MoteHue) [reply doc/source/operations/upgrading-openstack.rst:73]: We already have this at the bottom of the upgrading doc

**Comment by MoteHue** [comment]
Thanks @priteau, I was hoping for someone to spot anything I'd missed :)

File: doc/source/operations/upgrading-openstack.rst [hunk 2]
```diff
@@ -52,7 +52,38 @@ driver. Instructions for enabling the driver can be found `here
...
+
+   kayobe overcloud host command run --command "rm -rf /etc/kolla/heat-api /etc/kolla/heat-api-cfn /etc/kolla/heat-engine " --limit controllers -b
```
Comment (priteau) [inline doc/source/operations/upgrading-openstack.rst]: Nit: extra space at the end of the command.

**Review by priteau — CHANGES_REQUESTED** [review]

File: doc/source/operations/upgrading-openstack.rst [hunk 3]
```diff
@@ -52,7 +52,38 @@ driver. Instructions for enabling the driver can be found `here
...
+Then from the OpenStack CLI:
+
+.. code-block:: console
+
+   openstack service delete heat
+   openstack user delete heat
+   openstack domain set --disable heat_user_domain
+   openstack domain delete heat_user_domain
+
+You can drop the database too, unless you want to keep historical content.
```
Comment (priteau) [inline doc/source/operations/upgrading-openstack.rst]: We should say the ``heat`` database here to be sure no one misunderstands it. And provide a command line example?

**Review by priteau — CHANGES_REQUESTED** [review]

File: doc/source/operations/upgrading-openstack.rst (hunk 3 above)
Comment (MoteHue) [reply doc/source/operations/upgrading-openstack.rst]: That would be an unfortunate misunderstanding.

**Review by priteau — APPROVED** [review]

**Pull request merged**